test/Npgsql.Tests/CommandBuilderTests.cs

```diff
@@ -444,6 +444,79 @@ public void DeriveParametersFunctionReturningSetofRecord()
         }
 
         #endregion
+
+        #region CommandType.Text
+
+        [Test, Description("Tests parameter derivation for parameterized queries (CommandType.Text)")]
+        public void DeriveParametersCommandTypeTextOneParameterWithSameType()
+        {
+            using (var conn = OpenConnection())
+            {
+                conn.ExecuteNonQuery("CREATE TEMP TABLE mytable(id int, val text);");
+                var cmd = new NpgsqlCommand(
+                    @"INSERT INTO mytable VALUES(:x, 'some value');
+                    UPDATE mytable SET val = 'changed value' WHERE id = :x;
+                    SELECT val FROM mytable WHERE id = :x;",
+                    conn)
+                { CommandType = CommandType.Text };
+                NpgsqlCommandBuilder.DeriveParameters(cmd);
+                Assert.That(cmd.Parameters, Has.Count.EqualTo(1));
+                Assert.That(cmd.Parameters[0].Direction, Is.EqualTo(ParameterDirection.Input));
+                Assert.That(cmd.Parameters[0].ParameterName, Is.EqualTo(":x"));
```

Comment:
IMHO the parameter names should be derived without the special placeholder character (i.e. `x` and not `:x`) - the character is a mechanism for *identifying* the placeholder in the SQL text, and there's no reason to have it in the parameter name. Any objection?

Reply:
No objection. It was this way for functions and I think we need to be consistent. But I don't mind about the concrete way of consistency.

Reply:
OK, then let's drop the special placeholder character from derived parameter names.

Reply:
While testing SqlClient today I actually found out that it keeps the @ sign in the parameter name - just in case you want to rethink this because of better compatibility/portability.

Reply:
Personally I still think we should discard the `@` and the `:` as PostgreSQL doesn't have it in function names anyways and one probably shouldn't have any assumptions for derived parameter names.

Reply:
Interesting about SqlClient, but it doesn't really change my mind - parameter placeholder details (correctly!) seem to be a completely database-dependent in ADO.NET, since every database supports them differently. I also can't really imagine a case where compatibility/portability would be hurt by omitting the character, application code should not contain assumptions on how the parameter name is derived etc.

So let's leave the character out.